pandas/tests/series/test_dtypes.py

```diff
@@ -37,6 +37,12 @@ def test_astype(self, dtype):
         assert as_typed.dtype == dtype
         assert as_typed.name == s.name
 
+    def test_asobject_deprecated(self):
+        s = Series(np.random.randn(5), name='foo')
+        with tm.assert_produces_warning(FutureWarning):
+            o = s.asobject
```

Comment:
you can assert inside the with

Reply:
I don't see why: by having it outside, we're clear that the with statement is about the call ``s.asobject`` and not the call's return value.

Reply:
to be honest I actually don't care about the return value here, this is adequately tested elsewhere, the point of this test is just to assert the warning.